pkg/cli/cmd/resource/delete/delete.go

```diff
@@ -169,3 +182,63 @@ func (r *Runner) Run(ctx context.Context) error {
 
 	return nil
 }
+
+func (r *Runner) extractEnvironmentAndApplicationIDs(ctx context.Context, client clients.ApplicationsManagementClient) (environmentID resources.ID, applicationID resources.ID, err error) {
+	resource, err := client.GetResource(ctx, r.ResourceType, r.ResourceName)
+	if err != nil {
+		return resources.ID{}, resources.ID{}, err
+	}
+
+	// Note: The following cases are all possible:
+	//
+	// 1. The resource has an environment and an application. (common case for a standard resource)
+	// 2. The resource has an environment but no application. (possible case for a *shared* standard resource)
+	// 3. The resource has an application but no environment. (common case for a *core* resource like a container)
+	//		- In this case, the environment can be looked up through the application
+	//		- See: https://github.com/radius-project/radius/issues/2928
+	if resource.Properties["environment"] != nil {
+		environmentID, err = convertToResourceID(resource.Properties["environment"])
+		if err != nil {
+			return resources.ID{}, resources.ID{}, err
+		}
+	}
+
+	if resource.Properties["application"] != nil {
+		applicationID, err = convertToResourceID(resource.Properties["application"])
+		if err != nil {
+			return resources.ID{}, resources.ID{}, err
+		}
+	}
+
+	// At this point we have the environment and application IDs **if** they were returned by
+	// the API. That covers case 1 & 2. Now we need to handle case 3, by doing an additional
+	// lookup.
+	if !environmentID.IsEmpty() {
```

Comment:
Can the **applicationID** be also empty?

Reply:
Yes, this is valid (case 2 above).